### PR TITLE
Link pull request to test results

### PR DESCRIPTION
**Pull request by Allda**
Test result produced by a preflight are linked with a current pull
request that is processed by the hosted pipeline.

JIRA: ISV-925

### CODE REVIEW

File: ansible/roles/operator-pipeline/templates/openshift/tasks/preflight.yml [hunk 1]
```diff
@@ -24,6 +24,7 @@ spec:
           echo "Preflight results already exists- skipping"
           echo -n "none" | tee $(results.log_output_file.path)
           echo -n "none" | tee $(results.result_output_file.path)
+          echo -n "none" | tee $(results.artifacts_output_dir.path)
```
[the remark this answers is not in the document]
Comment (Allda) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/preflight.yml:27]: Yes, the pipeline was failing when preflight results already exist

File: ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml [hunk 2]
```diff
@@ -440,6 +440,29 @@ spec:
...
+        name: link-pull-request
+      params:
+        - name: test_result_id
+          value: "$(tasks.get-ci-results.results.test_result_id)"
+        - name: pyxis_url
+          value: "$(params.pyxis_url)"
+        - name: pyxis_cert_path
+          value: /workspace/pyxis-ssl-credentials/operator-pipeline.pem
+        - name: pyxis_key_path
+          value: /workspace/pyxis-ssl-credentials/operator-pipeline.key
```
[the remark this answers is not in the document]
Comment (Allda) [reply ansible/roles/operator-pipeline/templates/openshift/pipelines/operator-hosted-pipeline.yml:457]: This path + workspace is used across multiple tasks. I'll file a ticket to change it in all places.

File: ansible/roles/operator-pipeline/templates/openshift/tasks/link-pull-requst.yml [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    - name: pyxis_cert_path
+      default: ""
+      description: Path to Pyxis certificates. Valid only when internal Pyxis is used.
+    - name: pyxis_key_path
+      default: ""
+      description: Path to Pyxis key. Valid only when internal Pyxis is used.
+    - name: test_result_id
+    - name: pyxis_url
+    - name: pull_request_url
+    - name: pull_request_status
```
[the remark this answers is not in the document]
Comment (Allda) [reply ansible/roles/operator-pipeline/templates/openshift/tasks/link-pull-requst.yml]: Done

**Review by MarcinGinszt — APPROVED** [review]
LGTM

**Pull request merged**